rust/chains/hyperlane-ethereum/src/rpc_clients/trait_builder.rs

```diff
@@ -210,10 +211,19 @@ pub trait BuildableWithProvider {
         M: Middleware + 'static,
     {
         Ok(if let Some(signer) = signer {
-            let signing_provider = wrap_with_signer(provider, signer)
+            // - The signing provider is used for sending txs, which may end up stuck in the mempool due to
+            // gas pricing issues. So we first wrap the provider in a gas escalator middleware.
+            // - When txs reach the gas escalator, they will already have been signed by the signer middleware,
```

Comment:
nit: I think this comment is now stale -- iiuc when txs reach the gas escalator they have not been signed yet

Reply:
updated and also mentioned the placing of the nonce manager as the outermost middleware